src/vm/methodtable.cpp

```diff
@@ -2355,6 +2358,8 @@ bool MethodTable::ClassifyEightBytes(SystemVStructRegisterPassingHelperPtr helpe
     }
     CONTRACTL_END;
     
+    _ASSERTE(!isNativeStruct);
```

Comment:
From the HFA code, it looks like we should perhaps be falling back to `ClassifyEightBytesForManagedStruct` if `HasLayout()` is false (see `MethodTable::GetNativeHFAType()`: https://github.com/dotnet/coreclr/blob/master/src/vm/class.cpp#L1698).